snap/channel/channel.go

```diff
@@ -39,6 +39,40 @@ type Channel struct {
 	Branch       string `json:"branch,omitempty"`
 }
 
+func isSlash(r rune) bool { return r == '/' }
+
+func Canonize(s string) (string, error) {
+	if s == "" {
+		return "", nil
```

Comment:
This looks correct because we do have this empty values for TrackingChannel, for example when we firstboot with unasserted snaps.

It doesn't seem to be tested though.